cpu/atmega2560/periph/gpio.c

```diff
@@ -102,6 +102,24 @@ int gpio_init(gpio_t pin, gpio_dir_t dir, gpio_pp_t pullup)
     else {
         _SFR_MEM8(_ddr_addr(pin)) &= ~(1 << _pin_num(pin));
         res = bit_is_clear(_SFR_MEM8(_ddr_addr(pin)), _pin_num(pin));
+
+        if (res == 0) {
```

Comment:
@danielamkaer Yes, this is what the if statement says, but the bit_is_clear macro returns a value which is **not 0** when the bit is **clear**.
The if statement checks if the return value is **equal 0** and that says, the bit is **set**.

>The DDxn bit in the DDRx Register selects the direction of this pin. If DDxn is written logic one, Pxn is configured
as an output pin. **If DDxn is written logic zero, Pxn is configured as an input pin.**

The bit must be cleared for input pins.
You ask if the bit is set.

BTW: I can't initialize input pins with this, it returns always -1